### PR TITLE
Add a syllabus to guide people who are learning QMK

### PR DESCRIPTION
**Pull request by skullydazed**
Inspired by a conversation on discord I put together a syllabus for QMK. This can be used by people who want to learn QMK but aren't sure about what the right path to follow is.

## Types of Changes

- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/syllabus.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+* **Configuring IDEs** - Optional
+    * [Eclipse](other_eclipse.md)
+    * [VSCode](other_vscode.md)
+
+# Intermediate Topics
+
+These topics start dig into some of the features that QMK supports. You don't have to read all of these documents, but some of the documents in the Advanced Topics section won't make sense if you skip over some of these.
+
+* **Learn How To Configure Features**
+    * Configuration Overview <!-- FIXME(skullydazed/anyone): write this document -->
```
Comment (Erovia) [inline docs/syllabus.md:26]: I think we should either remove this until the document is not written or note it somehow, so the user will not find an "unclickable link".

Comment (Erovia) [inline docs/syllabus.md:26]: I think we should either remove this until the document is not written or note it somehow, so the user will not find an "unclickable link".

Comment (skullydazed) [reply docs/syllabus.md:26]: How about this?

```suggestion
    <!-- * Configuration Overview  FIXME(skullydazed/anyone): write this document -->
```

Comment (Erovia) [reply docs/syllabus.md:26]: :+1:

**Pull request merged**